wazuh/wazuh-dashboard-plugins
# Fix show call-out warning when there is no data in the vulnerabilities index

### Description
This pull request adds a vulnerability index data check and displays a warning call-out in case the index does not exist or there is no data in it. This applies to both the Dashboard tab and the Inventory tab.
This pull request also fixes the following changes:
- Since the architecture field will not be in the index, it is necessary to remove the architecture field - both tabs vuls
- Since the event field is not going to be in the index, it is necessary:
  - `@timestamp` is used instead of `event.create` in the "Accumulation of the most detected vulnerabilities" visualization
  - Delete `event.create` in inventory
- Capitalize is used on the values in the `vulnerability.severity` fields.
- In the OS filters of the vulnerabilities dashboard, use the `os.full` field
- Added WarningError to error handling to recognize vulnerability warnings and adjust the health check accordingly
 
### Issues Resolved
- #6166 

### Evidence
#### Health check warning message when no indices match vulnerability index pattern 

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/9e26ac01-ea20-4d62-a52e-5242dad962c6)

#### Vulnerability detection dashboard when no indices match vulnerability index pattern 

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/32658dd2-9ade-4a18-8aca-e289865a54eb)

#### Vulnerability detection inventory when no indices match vulnerability index pattern 

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/7962886e-e3b7-4be6-a05e-55be5d489419)

#### Health check when the vulnerability index exists
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/c7009810-fe88-4cea-b899-8a0be21d0517)

#### Vulnerability detector dashboard when the vulnerability index exists, but mapping is wrong
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/b8398304-f50a-4edc-a280-af30001c89d4)

#### Vulnerability detector dashboard when the vulnerability index exists and mapping is correct
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/7d4c805f-0da1-4f11-9e49-8ddcece28fb0)

#### Vulnerability detector inventory when the vulnerability index exists and mapping is correct
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/c231cf5d-dfdc-4068-b57c-96e44f8b0bf6)


### Test
Before remember to create the mapping of port `9200:9200` in `osd1` in the `dev.yml`

- When raising the environment for the first time. If the vulnerability index does not exist, a Warning should appear in the Health Check. But it should allow you to continue. When going to the vulnerabilities detector (Dashboard and Inventory), Warning should appear because no indices match vulnerability index pattern.
- Create the index (wazuh-states-vulnerabilities) in the manager and insert data (using script of [comment](https://github.com/wazuh/wazuh-dashboard-plugins/issues/5763#issuecomment-1779564103)). The health-check will correctly create the index pattern and pass the health-check, although the dashboard will not be seen completely, because mapping is wrong.
- Insert the corresponding mapping and see that the healthcheck passes correctly and the dashboard is shown complete (using dev-tools of [comment](https://github.com/wazuh/wazuh-dashboard-plugins/issues/5763#issuecomment-1779564103)). (Refresh vulnerabilities index-pattern then). The Inventory tab should also appear complete.


### Check List
